src/radical/entk/appman/wfprocessor.py

```diff
@@ -353,12 +353,19 @@ def _update_dequeued_task(self, deq_task):
                     if stage.post_exec:
                         self._execute_post_exec(pipe, stage)
 
-                    # If pipeline has completed, make state
-                    # change
-                    if pipe.completed:
+                    # if pipeline got suspended, advance state accordingly
+                    if pipe.state == states.SUSPENDED:
+                      self._advance(pipe, 'Pipeline', states.SUSPENDED)
+
+                    else:
+                        # otherwise perform normal stage progression
+                        pipe._increment_stage()
```

Comment:
This will create a double pipeline increment with line 406. I am not sure how to fix it.